### PR TITLE
ci: update renovate process [skip chromatic]

### LINKED ISSUE #914: ci: 👷 refactor dependency update process
## User Story
As a developer of the Solid Design System, I would like to rely on an (as far as possible) automated process for security and standard updates, and also have manual update tasks planned as normal iteration tasks on the project board. This should give the team more focus and clarity who is taking over and no updates will be missed.
### Suggested Solution

- [x] reconfigure renovate to only create PRs and auto-merge security issues
- [x] add an exception to the chromatic pipeline ignoring renovate PRs (Security is more important than screenshot testing)
- [x] update the developers guide to the new concept:
1. renovate takes care on security issues
2. renovate creates the "Dependency Dashboard" ticket
3. the team plans and assignes the dashboard ticket to a specifc team member every iteration 
4. the assignee runs a `pnpm update all`, creates a respective PR within the iteration and closes the ticket

- [x] a evaluation ticket has to be created and added to iteration 10, in order to have a quick retro on how the new process works for the team. => #1052 

## Environment (GitHub Actions or Azure DevOps)
GitHub
## DoR
- [x] Item has business value
- [x] Item has been estimated by the team
- [x] Item is clear and well-defined
- [x] Item dependencies have been identified

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Implementation works successfully on `feature` branch

### PR DESCRIPTION
**Pull request by karlbaumhauer**
## Description:
- removes all automatic updates from renovate config unless its a security update and add custom PR title prefix
- adds branch exception to chromatic workflow to ignore security updates from renovate
- adds a new chapter to the developer guide to explain the new procedure

closes #914

## Definition of Reviewable:
- [x] Documentation is created/updated
- [x] Migration Guide is created/updated
- [x] relevant tickets are linked

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -1,37 +1,34 @@
...
-    },
-    {
-      "matchPackagePrefixes": ["@types/"],
-      "automerge": true,
-      "major": {
-        "automerge": false
-      }
-    }
+  "matchUpdateTypes": [
+    "security"
```
Comment (Vahid1919) [inline renovate.json:11]: I don't think this is a valid type (https://docs.renovatebot.com/configuration-options/#matchupdatetypes)

**Review by Vahid1919 — CHANGES_REQUESTED** [review]

**Comment by Vahid1919** [comment]
@karlbaumhauer I found these resources here about vulnerability updates:

- https://github.com/renovatebot/renovate/discussions/15490
- https://docs.renovatebot.com/configuration-options/#vulnerabilityalerts

Let me know if I can help

File: renovate.json (hunk 1 above)
Comment (karlbaumhauer) [reply renovate.json:11]: makes sense... even though I found this solution as well. But lets stick to the official documentation.

File: renovate.json [hunk 2]
```diff
@@ -1,37 +1,18 @@
...
-    }
-  ],
-  "lockFileMaintenance": {
-    "enabled": true,
-    "automerge": false
-  },
-  "major": {
-    "automerge": false
+  "rebaseWhen": "never",
+  "vulnerabilityAlerts": {
```
Comment (Vahid1919) [inline renovate.json:11]:
```suggestion
  "vulnerabilityAlerts": {
  "enabled": true,
```

adding enables makes sure that dependency updates that we disabled aren't blocked if they have security updates.

File: renovate.json [hunk 3]
```diff
@@ -1,37 +1,18 @@
...
+    "labels": [
+      "npm dependencies, renovate, security"
+    ],
+    "automerge": true
   },
```
Comment (Vahid1919) [inline renovate.json:16]:
```suggestion
  },
  "osvVulnerabilityAlerts": true,
```
I would also like to include this for additional vulnerability checks. (https://docs.renovatebot.com/configuration-options/#osvvulnerabilityalerts)

**Review by Vahid1919 — APPROVED** [review]
Besides the suggested changed, LGTM!

File: renovate.json (hunk 2 above)
Comment (karlbaumhauer) [reply renovate.json:11]: good point. don't know why I missed this egain... ;-) thx.

File: renovate.json (hunk 3 above)
Comment (karlbaumhauer) [reply renovate.json:16]: good point. thx.

**Review by mariohamann — APPROVED** [review]

**Pull request merged**